src/main/java/uk/gov/hmcts/reform/fact/data/api/services/CourtOpeningHoursService.java

```diff
@@ -0,0 +1,178 @@
+package uk.gov.hmcts.reform.fact.data.api.services;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.fact.data.api.entities.OpeningHourType;
+import uk.gov.hmcts.reform.fact.data.api.entities.Court;
+import uk.gov.hmcts.reform.fact.data.api.entities.CourtCounterServiceOpeningHours;
+import uk.gov.hmcts.reform.fact.data.api.entities.CourtOpeningHours;
+import uk.gov.hmcts.reform.fact.data.api.entities.types.DayOfTheWeek;
+import uk.gov.hmcts.reform.fact.data.api.errorhandling.exceptions.CourtResourceNotFoundException;
+import uk.gov.hmcts.reform.fact.data.api.repositories.CourtCounterServiceOpeningHoursRepository;
+import uk.gov.hmcts.reform.fact.data.api.repositories.CourtOpeningHoursRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@Slf4j
+public class CourtOpeningHoursService {
+
+    private final CourtOpeningHoursRepository courtOpeningHoursRepository;
+    private final CourtCounterServiceOpeningHoursRepository courtCounterServiceOpeningHoursRepository;
+    private final CourtService courtService;
+    private final OpeningHoursTypeService openingHoursTypeService;
+
+    public CourtOpeningHoursService(
+        CourtOpeningHoursRepository courtOpeningHoursRepository,
+        CourtCounterServiceOpeningHoursRepository courtCounterServiceOpeningHoursRepository,
+        CourtService courtService,
+        OpeningHoursTypeService openingHoursTypeService) {
+        this.courtOpeningHoursRepository = courtOpeningHoursRepository;
+        this.courtCounterServiceOpeningHoursRepository = courtCounterServiceOpeningHoursRepository;
+        this.courtService = courtService;
+        this.openingHoursTypeService = openingHoursTypeService;
+    }
+
+    /**
+     * Get a list of opening hours by court ID.
+     *
+     * @param courtId The court ID to find the opening hours for.
+     * @return The list of opening hours record or 204 if the court has no opening hours.
+     * @throws CourtResourceNotFoundException  if no opening hours record exists for the court.
+     */
+    public List<CourtOpeningHours> getOpeningHoursByCourtId(UUID courtId) {
+        var openingHours = courtOpeningHoursRepository.findByCourtId(courtService.getCourtById(courtId).getId());
```

Comment:
If openingHours is optional, you can do this in one line I think. I.e. courtOpeningHoursRepository.findByCourtId(courtService.getCourtById(courtId).getId()).elsethrows then the rest

I believe this may be a team preference but I think if we don't use var but specify the type it makes it easier for readability and also for PRs too. As I would know automatically if it were an optional haha